securedrop/journalist_app/api2.py

```diff
@@ -0,0 +1,378 @@
+"""
+This file specifies the synchronization strategy for the v2 Journalist API,
+including:
+
+1. A semi-`literate <https://en.wikipedia.org/wiki/Literate_programming>`_
+   reference implementation in Python of the structures and algorithms for
+   versioning and diffing resources, which can be easily replicated in another
+   language (e.g., TypeScript).  You can view this documentation with::
+
+       $ make docs  # then see "build/docs/"
+
+2. An initial set of test vectors.  To keep this file self-contained and
+   self-testing, these are implemented here as doctests, but they should also be
+   easily replicated in another language.  You can run these tests inside the
+   development shell with::
+
+       $ python -m doctest journalist_app/api2.py
+
+3. A scaffold (i.e., schemas and stubs) for the endpoints the new API provides.
+   Most raise ``NotImplementedError``; a few are implemented for demonstration.
+   You can view the OpenAPI specification that flask-smorest generates from this
+   scaffold by running ``make dev``, logging into the Journalist Interface, and
+   navigating to <http://localhost:8081/docs>.
+
+   The OpenAPI specification can also be used to generate TypeScript types
+   including JSON Schema validation (via openapi-typescript) and/or full API
+   clients (via openapi-generator).
+
+
+Overview
```

Comment:
@legoktm and I concluded that it isn't worth adding Sphinx here just to keep this documentation self-contained in a single file.  I'll make `api2.py` a subpackage and make this module docstring its readme.

Reply:
Done in 7113f0e4c7a2abe6a1142fbede18f704d0d5a94a.